### PR TITLE
Feat add auth functionality

### LINKED ISSUE #7: Add Auth functionality
## To be added:
- Auth context
- Auth page

### PR DESCRIPTION
**Pull request by kjais1720**
PR for Issue #7 (closes issue #7 )

## Overview
- Created Auth Context and Auth modal
- Created Auth middleware
- Created Loaders and Toast
- Updated header
- Updated `useAxios`

## Preview
image

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,13 +1,18 @@
 import "./App.css";
 import { Header, Footer } from "components";
-import {AllRoutes} from "./AllRoutes";
-
+import { AllRoutes } from "./AllRoutes";
+import { useAuth } from "contexts";
+import { Toast } from "components";
```
Comment (rahulcs754) [inline src/App.js:5]: you can access in toast line number 2 .
I think this line is no need

**Review by rahulcs754 — APPROVED** [review]
The code looks clean and good.
Provide some suggestion

File: src/App.js (hunk 1 above)
Comment (kjais1720) [reply src/App.js:5]: Thanks for pointing it out, I have fixed it in the latest commit

**Review by aishwarya-bm — APPROVED** [review]
Looks great with modular & clean code, good naming conventions.  I like how you defined each parameter in functions.

**Pull request merged**